lib/fluent/plugin/output.rb

```diff
@@ -310,6 +310,10 @@ def configure(conf)
             log.warn "'flush_at_shutdown' is false, and buffer plugin '#{buf_type}' is not persistent buffer."
             log.warn "your configuration will lose buffered data at shutdown. please confirm your configuration again."
           end
+
+          if @flush_mode != :interval && buffer_conf.has_key?('flush_interval')
```

Comment:
If we doesn't take care about compatibility layer, it's better to raise configuration errors when both of `flush_mode lazy` and `flush_interval` are specified in configuration file.
But compatibility layer may specify these both if a plugin is TimeSlicedOutput and `flush_interval` exists in configuration. So we can't raise errors for such cases. `warn` is better.